scripts/teamd_increase_retry_count.py

```diff
@@ -0,0 +1,236 @@
+        MACField("actor_system", None),
+        ShortField("actor_key", 0),
+        ShortField("actor_port_priority", 0),
+        ShortField("actor_port_number", 0),
+        ByteField("actor_state", 0),
+        XStrFixedLenField("actor_reserved", "", 3),
+        ByteField("partner_type", 2),
+        ByteField("partner_length", 20),
+        ShortField("partner_system_priority", 0),
+        MACField("partner_system", None),
+        ShortField("partner_key", 0),
+        ShortField("partner_port_priority", 0),
+        ShortField("partner_port_number", 0),
+        ByteField("partner_state", 0),
+        XStrFixedLenField("partner_reserved", "", 3),
+        ByteField("collector_type", 3),
+        ByteField("collector_length", 16),
+        ShortField("collector_max_delay", 0),
+        XStrFixedLenField("collector_reserved", "", 12),
+        ConditionalField(ByteField("actor_retry_count_type", 0x80), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("actor_retry_count_length", 4), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("actor_retry_count", 0), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(XStrFixedLenField("actor_retry_count_reserved", "", 1), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("partner_retry_count_type", 0x81), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("partner_retry_count_length", 4), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(ByteField("partner_retry_count", 0), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(XStrFixedLenField("partner_retry_count_reserved", "", 1), lambda pkt:pkt.version == 0xf1),
+        ByteField("terminator_type", 0),
+        ByteField("terminator_length", 0),
+        ConditionalField(XStrFixedLenField("reserved", "", 42), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(XStrFixedLenField("reserved", "", 50), lambda pkt:pkt.version != 0xf1),
+    ]
+
+split_layers(scapy.contrib.lacp.SlowProtocol, scapy.contrib.lacp.LACP, subtype=1)
+bind_layers(scapy.contrib.lacp.SlowProtocol, LACPRetryCount, subtype=1)
+
+def getPortChannelConfig(portChannelName):
+    process = subprocess.run(["teamdctl", portChannelName, "state", "dump"], capture_output=True)
+    return json.loads(process.stdout)
+
+def getLldpNeighbors():
+    process = subprocess.run(["lldpctl", "-f", "json"], capture_output=True)
+    return json.loads(process.stdout)
+
+def craftLacpPacket(portChannelConfig, portName, isResetPacket=False, newVersion=True):
+    portConfig = portChannelConfig["ports"][portName]
+    actorConfig = portConfig["runner"]["actor_lacpdu_info"]
+    partnerConfig = portConfig["runner"]["partner_lacpdu_info"]
+    l2 = Ether(dst="01:80:c2:00:00:02", src=portConfig["ifinfo"]["dev_addr"], type=0x8809)
+    l3 = scapy.contrib.lacp.SlowProtocol(subtype=0x01) 
+    l4 = LACPRetryCount()
+    if newVersion:
+        l4.version = 0xf1
+    else:
+        l4.version = 0x1
```

Comment:
This will override any version other than 0x1. For eg., when/if teamd follow version 0x2 for LACPDU, then this logic will forcefully set it to 0x1.

Reply:
If teamd does add support for version 0x2 LACPDUs and default to using that version, we would anyways need a transition mechanism of some kind to handle both 0x1 and 0x2 (assuming there's no built-in mechanism). I think if that does happen, we can add support for that here.